cpp/velox/compute/WholeStageResultIterator.cc

```diff
@@ -374,6 +365,23 @@ void WholeStageResultIterator::collectMetrics() {
 
   metrics_ = std::make_unique<Metrics>(statsNum);
 
+  const int64_t collectTaskStatsThreshold =
+      veloxCfg_->get<int64_t>(kTaskMetricsToEventLogThreshold, kTaskMetricsToEventLogThresholdDefault);
+  // Save and print the plan with stats if debug mode is enabled or showTaskMetricsWhenFinished is true.
+  if (collectTaskStatsThreshold >= 0 || veloxCfg_->get<bool>(kDebugModeEnabled, false) ||
+      veloxCfg_->get<bool>(kShowTaskMetricsWhenFinished, kShowTaskMetricsWhenFinishedDefault)) {
+    auto planWithStats = velox::exec::printPlanWithStats(*veloxPlan_.get(), taskStats, true);
+    std::ostringstream oss;
+    oss << "Native Plan with stats for: " << taskInfo_;
+    oss << "\n" << planWithStats << std::endl;
+    const auto& stats = oss.str();
+    if (static_cast<int64_t>(taskStats.terminationTimeMs - taskStats.executionStartTimeMs) >
```

Comment:
When `collectTaskStatsThreshold = -1` and `kDebugModeEnabled=true`, `stats` is also assigned. Is this expected?

Reply:
This is the original logic. The task stats will be printed if spark.gluten.sql.debug=true